Shared/Samples/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,273 @@
+                contentsOf: [
+                    observeAutoPanTask,
+                    observeLocationsTask,
+                    observeSatellitesTask
+                ]
+            )
+        }
+        
+        /// A detached task observing location display autoPan changes.
+        var observeAutoPanTask: Task<Void, Never> {
+            Task.detached { [unowned self] in
+                defer { print("mode in locationDisplay.$autoPanMode task ending") }
+                for await mode in locationDisplay.$autoPanMode {
+                    await MainActor.run {
+                        isRecenterButtonDisabled = nmeaLocationDataSource == nil || mode == .recenter
+                    }
+                }
+            }
+        }
+        
+        /// A detached task observing location data source location changes.
+        var observeLocationsTask: Task<Void, Never> {
+            Task.detached { [unowned self] in
+                defer { print("location in nmeaLocationDataSource.locations task ending") }
+                for await location in nmeaLocationDataSource.locations {
+                    guard let nmeaLocation = location as? NMEALocation,
+                          nmeaLocationDataSource.status == .started else { return }
+                    let horizontalAccuracy = Measurement(
+                        value: nmeaLocation.horizontalAccuracy,
+                        unit: UnitLength.meters
+                    )
+                    let verticalAccuracy = Measurement(
+                        value: nmeaLocation.verticalAccuracy,
+                        unit: UnitLength.meters
+                    )
+                    let accuracyText = String(
+                        format: "Accuracy - Horizontal: %@; Vertical: %@",
+                        distanceFormatter.string(from: horizontalAccuracy),
+                        distanceFormatter.string(from: verticalAccuracy)
+                    )
+                    await MainActor.run {
+                        accuracyStatus = accuracyText
+                    }
+                }
+            }
+        }
+        
+        /// A detached task observing NMEA location data source satellite changes.
+        var observeSatellitesTask: Task<Void, Never> {
+            Task.detached { [unowned self] in
+                defer { print("satellites in nmeaLocationDataSource.satellites task ending") }
+                for await satellites in nmeaLocationDataSource.satellites {
+                    guard nmeaLocationDataSource.status == .started else { return }
+                    
+                    // Update the satellites info status text.
+                    let satelliteSystems = satellites.filter {
+                        $0.system != nil
+                    }
+                    let satelliteSystemsText = ListFormatter.localizedString(
+                        byJoining: Set(satellites.map(\.system!.label)).sorted()
+                    )
```

Comment:
Since the `satelliteSystems` is a non-nil version of the `satellites`, we can use it here as well. Also, please use the new formatting API.

https://goshdarnformatstyle.com/list-style/